packages/legend-query-builder/src/components/explorer/QueryBuilderExplorerPanel.tsx

```diff
@@ -298,7 +304,36 @@ const QueryBuilderExplorerContextMenu = observer(
             (childNode) =>
               !(childNode.type instanceof Class) &&
               childNode.mappingData.mapped,
-          );
+          )
+          .filter((childNode) => {
+            const propertyExpression =
+              buildPropertyExpressionFromExplorerTreeNodeData(
+                childNode,
+                queryBuilderState.explorerState,
+              );
+            if (
+              queryBuilderState.fetchStructureState.implementation instanceof
+              QueryBuilderTDSState
+            ) {
+              const currentProjectColumns =
```

Comment:
can't you use the `.find` function instead of map and if the projection column is found to match your condition then you filter that out. Prevents you from having to use `.every` and going through all the projection columns.

Reply:
True!